IanGRap/Squish
# Ui refactor

Contributors: Austin

Goals:
- Refactor the UI code so that shared logic is all contained to one file
- Reorganize the file structure with respect to the UI scripts
- Implement an option where each menu can use the ColorManger's color as the button colors

What Changed;
- UIController
    - The new file that contains the shared menu logic
- IMenuInput
    - An interface for menu input capturing scripts
- UIMatchMenuInput
    - Was UIMenuInput, renamed to UIMatchMenuInput, implements IMenuInput
- UILobbyMenuInput
    - Now implements IMenuInput Interface
- UITitleMenu, UIOptionsMenu, UIPauseMenu, UIScoreMenu, UIWinScreenMenu
    - All of these have the shared logic removed and inherit from UIController
- All UI code was moved into a new folder called Assets/Scripts/UI

How to Test:
- Make sure all menus still work as expected.
- Make sure they also work in a build
- Make sure you can use ColorManager colors by changing Use Color Manager Colors on each menu
- Do you like the code changes? Any suggestions?
- Do you like the new folder structure?
